Sprint-3/revise/implement/password-validator.test.js

```diff
@@ -14,3 +14,58 @@ To be valid, a password must:
 
 You must breakdown this problem in order to solve it. Find one test case first and get that working
 */
+
+const {passwordValidator} = require('./password.validator');
+describe ("passwordValidator function", () => {
+
+//Typical cases
+
+test("should return false if input is less than 5 characters", () => {
+    const previousPasswords = ["passWord55@", "Access1!Point", "#MarioBros3"]
+    expect(passwordValidator("Tree", previousPasswords)).toBe(false);
+    expect(passwordValidator("1234", previousPasswords)).toBe(false);
+}); 
+
+test("should return false if input does not contain an uppercase letter", () => {
+    const previousPasswords = ["passWord55@", "Access1!Point", "#MarioBros3"]
+    expect(passwordValidator("watermelon", previousPasswords)).toBe(false);
+    expect(passwordValidator("56789", previousPasswords)).toBe(false);
+}); 
+
+test("should return false if input does not contain a lowercase letter", () => {
+    const previousPasswords = ["passWord55@", "Access1!Point", "#MarioBros3"]
+    expect(passwordValidator("TELEVISION", previousPasswords)).toBe(false);
+    expect(passwordValidator("01010101010", previousPasswords)).toBe(false);
```

Comment:
The test value "01010101010" is not specific enough. When this test fails, it could mean one or more of the followings:
- The function failed to check if the password contain a non-alphanumeric character
- The function failed to check if the password contain a lowercase letter
- The function failed to check if the password contain an uppercase letter

which is not exactly "... does not contain a lowercase letter".

Ideally, when a test fails, we want to know **exactly** why the test failed.
 
Many of the test values (invalid passwords) in this file are not specific enough.